components/num-util/src/fixed_decimal.rs

```diff
@@ -385,15 +385,13 @@ impl FromStr for FixedDecimal {
         if input_str == "" || input_str == "-" {
             return Err(Error::Syntax);
         }
-        let no_sign_str: &str;
         let mut is_negative = false;
-        if &input_str[0..1] == "-" {
+        let input_str = input_str.as_bytes();
+        let mut no_sign_str = input_str;
```

Comment:
why do you need to move it (copy?) and make it mutable?

Reply:
Fair point, fixed and removed mutability.